.github/workflows/dependabot-auto-merge.yml

```diff
@@ -6,10 +6,32 @@ permissions:
   pull-requests: write
 
 jobs:
-  auto_approval:
-    name: Auto approve qualifying PR
+  validate_this_is_dependabot:
     runs-on: ubuntu-latest
     if: ${{ github.actor == 'dependabot[bot]' }}
+    steps:
+      - run: echo "This PR was raised by Dependabot. Now let's see if we should auto-merge..."
+  validate_this_is_an_allowed_dependency:
+    needs: validate_this_is_dependabot
+    runs-on: ubuntu-latest
+    outputs:
+      is_allowed_dependency: ${{ steps.check_if_allowed_dependency.outputs.is_allowed_dependency }}
```

Comment:
Should this step `check_if_allowed_dependency` be specified in this file?
In your workflow code, I saw it is added https://github.com/ChrisBAshton/test-auto-merge/blob/356b073a3495922a783e2e56a4f9cb1e20825c51/.github/workflows/dependent-task.yml#L26